test/basic-querying.test.js

```diff
@@ -275,7 +289,7 @@ describe('basic-querying', function() {
       });
     });
 
-    it('should support "or" that is satisfied', function(done) {
+    bdd.itIf(!isCassandraConnector, 'should support "or" that is satisfied', function(done) {
```

Comment:
I am proposing:
 
 - add `connectorCapabilities.supportsOrOperator`
 - modify the cassandra connector to fail with a descriptive error when a filter with `or` is detected (see also https://github.com/strongloop/loopback-datasource-juggler/pull/1290#discussion_r109114458)